### PR TITLE
[docs] Improve Expo Router modals guide

### PR DESCRIPTION
**Pull request by amandeepmittal**
# Why

Fixes ENG-12963

Also: 
- Based on user feedback received about the native modal presentation behavior, explain that it is different for Android and iOS.
- Based on feedback from Expo Rouer team to include info about uses cases for React Native's `Modal` component and modal as a route

# How

Changes to the modals guide:
- Update the intro to explain there are two approach a dev can take to create a Modal inside their app (React Native `Modal` and modal screen)
- Add a section about React Native Modal to show a working demo on all platforms (Android, iOS, Web) and explain its use case
  - Based on feedback from Router team, add context that this approach should be used for most use cases
- Add a section to explain what a modal route is, show a working demo by adding a video and in what uses cases it should be used, with the following subsections to describe
  - Usage section on using modal routes for focused interactions in mobile apps.
  - An example setup with a Stack navigator to create modals and use the existing examples to organize content structure b creating new sub-sections.
  - Explain Modal presentation and dismiss behavior
  - Explain how to change status bar appearance on iOS
  - A section on `presentation` options available as additional information

# Test Plan

By running docs locally http://localhost:3002/router/advanced/modals/ or see Preview: /router/advanced/modals/.

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: docs/pages/router/advanced/modals.mdx [hunk 1]
```diff
@@ -1,79 +1,179 @@
...
+- On Android, the modal slides on top of the current screen. To dismiss it, use the back button to navigate back to the previous screen.
+- On iOS, the modal slides from the bottom of the current screen. To dismiss it, swipe it down from the top.
+- On web, the modal is presented as a separate route, and the dismiss behavior has to be provided manually using [`router.canGoBack()`](/router/navigating-pages/#imperative-navigation). Here's an example of how to dismiss the modal:
 
-```tsx app/modal.tsx
-import { View, Platform } from 'react-native';
-import { Link, router } from 'expo-router';
-import { StatusBar } from 'expo-status-bar';
+{/* prettier-ignore */}
+```tsx modal.tsx
```
Comment (Simek) [inline docs/pages/router/advanced/modals.mdx:109]:
```suggestion
```tsx app/modal.tsx
```

We call other snippets with modal examples as `app/modal.tsx`, let's retain that name also here.

File: docs/pages/router/advanced/modals.mdx [hunk 2]
```diff
@@ -1,79 +1,179 @@
...
+| --------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `card`                      | The new screen will be pushed onto a stack. The default animation on Android will vary depending on the OS version and theme. On iOS, it will slide from the side.               |
+| `modal`                     | The new screen will be presented modally, allowing for a nested stack to be rendered inside the screen.                                                                          |
+| `transparentModal`          | The new screen will be presented modally, with the previous screen remaining visible. This allows the content below to still be seen if the screen has a translucent background. |
+| `containedModal`            | On Android, fallbacks to `modal`. On iOS, uses `UIModalPresentationCurrentContext` modal style.                                                                                  |
```
Comment (Simek) [inline docs/pages/router/advanced/modals.mdx:176]:
```suggestion
| `containedModal`            | On Android, fallbacks to `modal`. On iOS, uses [`UIModalPresentationCurrentContext`](https://developer.apple.com/documentation/uikit/uimodalpresentationstyle/uimodalpresentationcurrentcontext) modal style.                                                                                  |
```

Can we link iOS presentation modes to Apple docs?

**Review by Simek — APPROVED** [review]
I really like the updates and clarifications! 🚀

**Comment by ningacoding** [comment]
It still needs improvements, I'm trying to use a modal but it always is a normal screen with black background.

docs example says it will be normal screen when reloading or entering directly to the route, but even with `router.pus('/modal')` its a black screen.

Code:
``` jsx
<Stack.Screen
              name="modal"
              options={{
                headerShown: false,
                presentation: 'fullScreenModal',
              }}
            />
```

Version:
`"expo-router": "~3.5.21",`

**Comment by amandeepmittal** [comment]
Hey @ningacoding, without seeing minimal reproducible example, I cannot tell what might be wrong. It'll be better if you can open a new GitHub issue with the example.

This document describes steps for creating a modal route, use cases when you can use the `Modal` component or a modal route, and available presentation options when using a modal as the route. It is difficult to take every use case into account and explain that in the document itself.

**Comment by ningacoding** [comment]
> Hey @ningacoding, without seeing minimal reproducible example, I cannot tell what might be wrong. It'll be better if you can open a new GitHub issue with the example.
> 
> This document describes steps for creating a modal route, use cases when you can use the `Modal` component or a modal route, and available presentation options when using a modal as the route. It is difficult to take every use case into account and explain that in the document itself.

Hi bro, it's a fresh install and new project with default broilerplate, just add that modal and navigate into the expo route.

It's a black screen, nothing similar to common modal look and feel.

**Comment by amandeepmittal** [comment]
@ningacoding, changing the presentation to `fullScreenModal` works on my end.


https://github.com/user-attachments/assets/03273003-1ffd-400f-b1db-312625adf438

When you say "It's a black screen, nothing similar to common modal look and feel.", I think you are referring to the behavior difference between how modal behaves on iOS vs other platforms, such as Android. The docs updates in this PR do mention that.

See this new video in the doc where it shows different default behavior of a modal route on Android, iOS, and Web: http://docs.expo.dev-pr-30766.s3-website-us-east-1.amazonaws.com/router/advanced/modals/#modal-screen-using-expo-router.

You might need to create a custom modal or use an external library if you're looking to achieve the similar behavior for iOS (for example: https://ui.gorhom.dev/components/bottom-sheet/).

**Comment by ningacoding** [comment]
> @ningacoding, when you say "It's a black screen, nothing similar to common modal look and feel.", I think you are referring to the behavior difference between how modal behaves on iOS vs other platforms, such as Android. The docs updates in this PR do mention that.
> 
> See this new video in the doc where it shows different default behavior of a modal route on Android, iOS, and Web: http://docs.expo.dev-pr-30766.s3-website-us-east-1.amazonaws.com/router/advanced/modals/#modal-screen-using-expo-router.
> 
> The differences between available `presentation` options for Android and iOS are described here: http://docs.expo.dev-pr-30766.s3-website-us-east-1.amazonaws.com/router/advanced/modals/#presentation-options.
> 
> You might need to create a custom modal or use an external library if you're looking to achieve the similar look and feel as of the iOS default modal behavior (for example: https://ui.gorhom.dev/components/bottom-sheet/) and might not need a modal route.
> 
> For the blank screen, try styling the modal route screen and see if the contents of the screen are shown.

I have an idea, as others old router libraries, for Android/Web a modal can just be a translucent or transparent background or some opacity, so we can see background (previous screen) over the content of the next screen, and we can just style the modal behavior as you suggested, BUT keeping the router benefits.

It's just an idea taken from old libraries, I really love this new expo router concept.

Thanks ❤️

**Comment by focux** [comment]
The web implementation seems to be missing some important details. While native platforms like iOS and Android offer a consistent and native modal experience, web modals typically appear centered with an opaque overlay rather than full-screen.

I'm currently migrating a large React Native app to Expo Router. Most of our modals are implemented as separate screens using React Navigation, primarily for the convenience of opening them through navigation. However, for the web version, we'd like to maintain this convenience without necessarily changing the URL, or at least have the option to choose whether the URL changes. I understand that in some cases, changing the URL for modals can be useful, so having this flexibility would be ideal.

I see that you're recommending the React Native Modal component, which could certainly help address some of these issues. However, I believe the convenience offered by screen-based modals makes them worth considering as well.
Additionally, it would be beneficial to have an option to render modals within deeply nested layouts, similar to a portal. This would allow the modal to be positioned absolutely in front of all content, rather than being framed within the current layout.

I'm not sure if this is the right place to raise these points, but I came across this PR while searching for more information about expo router modal implementation. Any suggestions would be greatly appreciated.

File: docs/pages/router/advanced/modals.mdx [hunk 3]
```diff
@@ -1,79 +1,199 @@
...
 import { ContentSpotlight } from '~/ui/components/ContentSpotlight';
 import { FileTree } from '~/ui/components/FileTree';
 
-Modals are a common pattern in mobile apps. They are a way to present a new screen on top of the current screen. They are often used for things like creating a new account, or for a user to select an option from a list.
+Modals are a common user interface pattern in mobile apps. They are used to present content on top of the existing screen and is used for different purposes, such as displaying confirmation alerts or standalone forms. You can create modals in your app using the following methods:
+
+- Use React Native's [`Modal`](https://reactnative.dev/docs/modal) component.
+- Use Expo Router's special file-based syntax to create a modal screen within the app's navigation system.
+
+Each approach has its specific use cases. Understanding when to use each method is important for creating a positive user experience.
```
Comment (amandeepmittal) [inline docs/pages/router/advanced/modals.mdx:14]:
```suggestion
Each approach has its specific use case. Understanding when to use each method is important for creating a positive user experience.
```

**Comment by amandeepmittal** [comment]
Hi @focux, thank you for your input! I think it would be great to open a GitHub discussion for this. The documentation here is designed to explain when to use React Native Modal or Expo Router modal route, based on common use cases. While some apps may have niche scenarios, it’s always valuable to explore which solution works best for them.

The behavior described here reflects the default settings. The goal of this doc is to offer guidance on the default behavior provided by the library and to outline the available settings and options. We aim to keep the focus on the broader use cases, rather than diving into very specific or niche scenarios.

**Pull request merged**